### PR TITLE
Add OpenTelemetry VCS attributes

### PR DESCRIPTION
**Pull request by renchap**
This allows knowing exactly what code is generating a given span, and your tooling can use this information to provide better context, in our case on errors with Datadog

### CODE REVIEW

**Review by Gargron — DISMISSED** [review]

File: config/initializers/opentelemetry.rb
Comment (ClearlyClaire) [inline config/initializers/opentelemetry.rb:72]: This should be `Mastodon::Version.source_base_url`

**Comment by renchap** [comment]
Please do not merge, I need to remove the draft status for the image to build, but I need to ensure they are set correctly before merging

**Comment by renchap** [comment]
This seems to work, from the resulting container:

```
$ export | grep SOURCE
export SOURCE_COMMIT='d951f82d859b3c8882e46638f586302a41838fb6'
```

**Comment by ClearlyClaire** [comment]
Note that this commit is not in any branch of the repo, so depending on how Datadog works, it may fail to reference it.

This is because this commit is the result of merging the PR head on `main`, as it's what `pull_request` actions run on.

**Comment by ClearlyClaire** [comment]
Yes, for nightlies this should be the `HEAD` of `main`, not an out-of-branch commit

**Review by ClearlyClaire — APPROVED** [review]

**Review by julianocosta89 — COMMENTED** [review]
I was about to suggest using the gem `opentelemetry/semantic_conventions`, but it seems that `VCS Repository Attributes` are not yet part of it.

So, it looks good to me! 🚀

**Pull request merged**
